src/views/TechnicalAssistance/AdminHome/CloseRequest.tsx

```diff
@@ -0,0 +1,289 @@
+  ModalRef,
+  ModalToggleButton
+} from '@trussworks/react-uswds';
+
+import UswdsReactLink from 'components/LinkWrapper';
+import PageHeading from 'components/PageHeading';
+import useMessage from 'hooks/useMessage';
+import CloseTrbRequestQuery from 'queries/CloseTrbRequestQuery';
+import ReopenTrbRequestQuery from 'queries/ReopenTrbRequestQuery';
+import {
+  CloseTrbRequest,
+  CloseTrbRequestVariables
+} from 'queries/types/CloseTrbRequest';
+import {
+  ReopenTrbRequest,
+  ReopenTrbRequestVariables
+} from 'queries/types/ReopenTrbRequest';
+
+import Breadcrumbs from '../Breadcrumbs';
+
+/**
+ * Close or Re-open a request
+ */
+function CloseRequest() {
+  const { t } = useTranslation('technicalAssistance');
+
+  const { id, activePage, action } = useParams<{
+    id: string;
+    activePage: string;
+    action: 'close-request' | 'reopen-request';
+  }>();
+  const history = useHistory();
+
+  const { message, showMessage, showMessageOnNextPage } = useMessage();
+
+  const requestUrl = `/trb/${id}/${activePage}`;
+
+  let actionText: 'actionCloseRequest' | 'actionReopenRequest';
+
+  if (action === 'close-request') {
+    actionText = 'actionCloseRequest';
+  } else {
+    actionText = 'actionReopenRequest';
+  }
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isSubmitting }
+  } = useForm({
+    defaultValues: {
+      text: ''
+    }
+  });
+
+  const [mutateClose] = useMutation<CloseTrbRequest, CloseTrbRequestVariables>(
+    CloseTrbRequestQuery
+  );
+
+  const [mutateReopen] = useMutation<
+    ReopenTrbRequest,
+    ReopenTrbRequestVariables
+  >(ReopenTrbRequestQuery);
+
+  // Confirm modal for closing a request
+  const confirmModalRef = useRef<ModalRef>(null);
+
+  const submitClose = handleSubmit(formData => {
+    mutateClose({
+      variables: {
+        input: {
+          id,
+          reasonClosed: formData.text,
+          notifyEuaIds: ['ABCD'], // todo
+          copyTrbMailbox: false // todo
+        }
+      }
+    })
+      .then(result => {
+        showMessageOnNextPage(
+          <Alert type="success" slim className="margin-top-3">
+            {t(`${actionText}.success`)}
+          </Alert>
+        );
+        history.push(`/trb/${id}/request`);
+      })
+      .catch(err => {
+        showMessage(
+          <Alert type="error" slim className="margin-top-3">
+            {t(`${actionText}.error`)}
+          </Alert>
+        );
+      });
+  });
+
+  const submitReopen = handleSubmit(formData => {
+    mutateReopen({
+      variables: {
+        input: {
+          trbRequestId: id,
+          reasonReopened: formData.text,
+          notifyEuaIds: ['ABCD'], // todo
+          copyTrbMailbox: false // todo
+        }
+      }
+    })
+      .then(result => {
+        showMessageOnNextPage(
+          <Alert type="success" slim className="margin-top-3">
+            {t(`${actionText}.success`)}
+          </Alert>
+        );
+        history.push(`/trb/${id}/request`);
+      })
```

Comment:
There needs to be checks on GraphQl errors here. Currently only network errors are being checked in the `catch` block.  It is possible to combine these checks I think* like so:

```
const [mutateClose, { error: mutationError }] = useMutation<
    CloseTrbRequest,
    CloseTrbRequestVariables
  >(CloseTrbRequestQuery);
  
  if (error) {
   showMessage(
          <Alert type="error" slim className="margin-top-3">
            {t(`${actionText}.error`)}
          </Alert>
        );
  }
  ```
 We can start a larger discussion about this as I think there a bit of grey area of our understanding of how apollo handles errors.  
 
 Per Apollo docs, the `mutationError` should contain both Graphql and Network errors.  This differs from the segmented errors in the `catch` and `then` blocks.  These return their respective error types
 
[https://www.apollographql.com/docs/react/data/error-handling/](https://www.apollographql.com/docs/react/data/error-handling/)